### ISSUE TITLE
Consider publishing desktop Preview annotation for common sourceSet.

### ISSUE THREAD

**Issue by IlyaGulya** [JetBrains/compose-jb#2045]
Hello!
Currently desktop preview rendering is possible only in `desktop` sourceSet because `ui-tooling-preview` publishes `androidx.compose.desktop.ui.tooling.preview.Preview` annotation only for `desktop` sourceSet.

However, I've been able to successfully use current Compose Multiplatform IDE Support plugin to render preview defined in common sourceSet.

You can check the example here:https://github.com/IlyaGulya/compose-jb/tree/example/common-desktop-preview/experimental/examples/falling-balls-mpp
I've moved `GamePreview` to common sourceSet and added stub `androidx.compose.desktop.ui.tooling.preview.Preview` annotation to trigger Compose Multiplatform IDE Support plugin to add gutter icon on `GamePreview` function.

**Comment by syt0r** [comment]
This would be really helpful. Also it would be nice to have a drop down in preview tab to select target platform/device for preview as well as other features from android preview like multiple previews, zoom, grid view, interactive mode, etc. Is it on roadmap somewhere? I can't find any information regarding plans

**Comment by hm-tamim** [comment]
Any update on this?

**Comment by pjBooms** [comment]
not yet

**Comment by SerggioC** [comment]
Any updates on this issue?

**Comment by anwarpro** [comment]
any update ?

**Comment by amirhossein320** [comment]
> It is good to have multiplatform Preview annotation for all target platforms in commonMain sourceSet.
> 
> Hello! Currently desktop preview rendering is possible only in `desktop` sourceSet because `ui-tooling-preview` publishes `androidx.compose.desktop.ui.tooling.preview.Preview` annotation only for `desktop` sourceSet.
> 
> However, I've been able to successfully use current Compose Multiplatform IDE Support plugin to render preview defined in common sourceSet.
> 
> You can check the example here:https://github.com/IlyaGulya/compose-jb/tree/example/common-desktop-preview/experimental/examples/falling-balls-mpp I've moved `GamePreview` to common sourceSet and added stub `androidx.compose.desktop.ui.tooling.preview.Preview` annotation to trigger Compose Multiplatform IDE Support plugin to add gutter icon on `GamePreview` function.
> 
> I think Compose MPP team should consider publishing `androidx.compose.desktop.ui.tooling.preview.Preview` annotation for `common` sourceSet instead of `desktop`.

and need to install Compose Multiplatform IDE Support

**Comment by senk0n** [comment]
It would be perfect to make use of Android Studio preview feature for composables from commonMain, at least for android target for now. So it will show how composable preview function from common code will look like on Android, in AS's preview ide view.

I assume it could be easier to do, than making a full spectrum of it's feature for Compose Multiplatform IDE plugin from scratch.
And maybe it's just a matter of configuration, because even AS's Layout Inspector works for kmp composables if you use androidx compose compiler in android target.

**Comment by Burtan** [comment]
I think this is very important to have @Preview working in shared compose ui code.

**Comment by anox1337** [comment]
I was wondering what the suggested workflow currently is for creating Composables without the Preview? Just good old fashioned rebuild or creating the composables within androidApp and then after finishing, copying them over to shared?

**Comment by SirFilbido** [comment]
Any updates on this issue?

**Comment by chrisjenx** [comment]
This is probably my top feature right now, everything else is pretty good state (at least less of a priority right now), without previews live updating in common editing/updating/building screens is really slow/painful, while we can put previews in androidDebug, the screen living in commonMain means we need to manually build and refresh vs "native" compose dev where previews living in the same srcSet auto compile as you edit them.

- As a stop gap, updating the KMM plugin to allow build and refresh on save from the common would be a really big intial win, means don't have to keep switching windows/tabs etc.

**Comment by bartekpacia** [comment]
This works in latest Fleet release ([link](https://www.jetbrains.com/help/kotlin-multiplatform-dev/whats-new-fleet-1-31.html#support-for-compose-multiplatform-preview-annotation)), but (at least in my experience) it's very slow (a refresh takes 5-10 sec on M1 macbook) and doesn't support all bells and whistles of `@Preview` from native Android. Still, good progress.

**Comment by chrisjenx** [comment]
> Even the preview workaround using androidMain do not work when the commonMain composable is loading a resource. I'm getting this exception `org.jetbrains.compose.resources.MissingResourceException: Missing resource with path: composeResources/mobileapp.composeapp.generated.resources/drawable/splash_bg.jpg` Any idea?

Yeah there is another ticket for this, the Preview plugin doesn't support common resources.

**Comment by igordmn** [comment]
Let's keep this as a meta issue for other children tasks:
- https://github.com/JetBrains/compose-multiplatform/issues/4869 - Reduce variety of Preview annotations
- https://github.com/JetBrains/compose-multiplatform/issues/5078 - Support common org.jetbrains.compose.ui.tooling.preview.Preview in IDEA and Android Studio

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-2045">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by acmpo6ou** [comment]
> Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-2045">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

But why move to YouTrack?
